### PR TITLE
Random Battles Fixes

### PR DESCRIPTION
**Pull request by ACakeWearingAHat**
fix water stab enforcement in doubles (now forces water stab OR hyper voice, no matter the format. Only affects doubles primarina.)
remove focus sash cinccino (since it's already item-hardcoded with tail slap in another place)
clean up unused hyper voice rejection
consolidate power construct with other rejected abilities
Prevent Triage from occurring on Synthesisless Comfey

### CODE REVIEW

File: data/random-teams.ts [hunk 1]
```diff
@@ -1129,8 +1126,6 @@ export class RandomTeams {
 					rejectAbility = !counter['Grass'];
 				} else if (ability === 'Own Tempo') {
 					rejectAbility = isDoubles;
-				} else if (ability === 'Power Construct') {
-					rejectAbility = true;
```
Comment (TheImmortal) [inline data/random-teams.ts:1133]: I mean, this was a placeholder for when it is released with DLC2 and it'd be rejected based on certain conditions (for example: 10% would have an Aura Break set).

File: data/random-teams.ts [hunk 2]
```diff
@@ -1348,7 +1343,8 @@ export class RandomTeams {
 			item = 'Leftovers';
 
 		// Better than Leftovers
-		} else if (isLead && !['Disguise', 'Sturdy'].includes(ability) && !hasMove['substitute'] && !counter['recoil'] && !counter['recovery'] && species.baseStats.hp + species.baseStats.def + species.baseStats.spd < 255 && !isDoubles) {
+		} else if (isLead && !['Disguise', 'Sturdy'].includes(ability) && !hasMove['substitute'] && !hasMove['tailslap'] && !counter['recoil'] && !counter['recovery'] &&
+			species.baseStats.hp + species.baseStats.def + species.baseStats.spd < 255 && !isDoubles) {
```
Comment (TheImmortal) [inline data/random-teams.ts:1347]: I don't think Sash Cinccino should be removed.

File: data/mods/gen7/random-teams.ts [hunk 3]
```diff
@@ -116,6 +116,9 @@ export class RandomGen7Teams extends RandomTeams {
 				case 'icebeam':
 					if (hasAbility['Tinted Lens'] && !!counter['Status']) rejected = true;
 					break;
+				case 'morningsun':
+					if (counter.damagingMoves.length < 1) rejected = true;
+					break;
```
Comment (TheImmortal) [inline data/mods/gen7/random-teams.ts:121]: What is this for?

File: data/random-teams.ts (hunk 2 above)
Comment (TheImmortal) [reply data/random-teams.ts:1347]: At least not with a really specific hardcode like this.

Comment (ACakeWearingAHat) [reply data/random-teams.ts:1347]: i mean, you already hardcoded it to give it Life Orb

File: data/mods/gen7/random-teams.ts (hunk 3 above)
Comment (ACakeWearingAHat) [reply data/mods/gen7/random-teams.ts:121]: stealth rock morning sun dual screens solrock exists in gen 7

File: data/random-teams.ts (hunk 2 above)
Comment (ACakeWearingAHat) [reply data/random-teams.ts:1349]: and it's reported at least once a week. But you can remove it if you want.

File: data/mods/gen5/random-teams.ts [hunk 4]
```diff
@@ -456,7 +456,7 @@ export class RandomGen5Teams extends RandomGen6Teams {
 				} else if (ability === 'Sturdy') {
 					rejectAbility = (!!counter['recoil'] && !counter['recovery']);
 				} else if (ability === 'Swarm') {
-					rejectAbility = !counter['Bug'];
+					rejectAbility = (!counter['Bug'] || abilities.includes('Hustle'));
```
Comment (TheImmortal) [inline data/mods/gen5/random-teams.ts:459]: This is the wrong fix because Hustle has a much higher rating. I found the problem.

**Comment by TheImmortal** [comment]
I'm going to close this because almost all of it is included in my next update.

**Pull request closed without merging**